suite-common/connect-init/src/connectInitThunks.ts

```diff
@@ -134,17 +134,25 @@ export const connectInitThunk = createThunk<void, ConnectInitHooks | void, void>
                 const original: any = TrezorConnect[key as ConnectKey];
                 if (!original) return;
                 (TrezorConnect[key as ConnectKey] as any) = async (params: any) => {
-                    dispatch(lockDevice(true));
-                    const result = await synchronize(() => original(params));
-
-                    dispatch(lockDevice(false));
-                    dispatch(
-                        deviceActions.removeButtonRequests({
-                            // todo: device not 'thread safe' - meaning that device to which button requests have been added to might not
-                            // be the same re-selected device from this line. We should reuse device from params.
-                            device: selectSelectedDevice(getState()),
-                        }),
-                    );
+                    const result = await synchronize(async () => {
+                        const infoResult = await original({ ...params, __info: true });
+
+                        if (infoResult.success && infoResult.payload.useDevice) {
+                            dispatch(lockDevice(true));
+                        }
+                        return original(params).finally(() => {
+                            if (infoResult.success && infoResult.payload.useDevice) {
+                                dispatch(lockDevice(false));
+                                dispatch(
+                                    deviceActions.removeButtonRequests({
```

Comment:
I am thinking about registering listener for button requests here instead of doing it globaly to handle adding button requests. Only when method is expected to use device.